ansible/roles/jenkins-workspace/tasks/main.yml

```diff
@@ -106,3 +106,31 @@
   command: "git config gc.auto 0"
   args:
     chdir: "~binary_tmp/binary_tmp.git/"
+
+- name: Add nodesource signing key
+  apt_key:
+    url: https://deb.nodesource.com/gpgkey/nodesource.gpg.key
+    state: present
+
+- name: Add nodesource repo
+  apt_repository:
+    repo: deb https://deb.nodesource.com/node_12.x xenial main
+    state: present
+
+- name: Install node
+  package:
+    name: nodejs
+    state: present
+
+- name: Upgrade pip2
```

Comment:
I was getting errors with `make lint-py-build` on both Python 2 and 3, this upgrade seems to have fixed it. I'm hoping this is the right thing to do. Can anyone tell me what would happen if there's a python-pip or python3-pip update that comes in via apt? Would this upgrade get written over and need to be run again?

Reply:
`pip` _should_ point to `pip2` but you can be more assured by using `pip2` instead of `pip`.

In situations where I need absolute clarity, I tend to run pip as a module from the Python that I care about as in:
```
python3 -m pip install --upgrade pip
python3 -m pip install flake8
# or even better if multiple Python 3 instances might be installed on a single box...
python3.8 -m pip install --upgrade pip
python3.8 -m pip install flake8
```

Reply:
:thumbsup: switched to `executable: pip2`, I'll trust Ansible to get it right beyond that and the playbook runs with this change.